torch/_inductor/ir.py

```diff
@@ -3505,11 +3505,13 @@ def realize(self):
         self.data.name = V.graph.register_buffer(self.data)
         return self.data.name
 
-    def realize_hint(self):
+    def realize_hint(self, is_output=False):
         """
         Called on buffers we expect to be forced to realize later.
         """
-        if isinstance(self.data, (Pointwise, Reduction)) and self.num_reads() > 1:
```

Comment:
Wait.... why is this `num_reads`? Why would we want to avoid realizing a *buffer* that has 1 read? Shouldn't we want to avoid realizing a buffer that is read *from* only once?

Reply:
I guess we want to avoid realizing it because if it's just 1 read, recomputing or realizing is the same in terms of global memory reads (not quite, due to potentially different dtype, but close enough). It's all part of our recompute/realize heuristics that's pretty arbitrary, but some things are unambiguous, like once you computed output, write it out, you have to anyway.